zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -1221,36 +1221,23 @@ protected Observer makeObserver(FileTxnSnapLog logFactory) throws IOException {
     }
 
     @SuppressWarnings("deprecation")
-    protected Election createElectionAlgorithm(int electionAlgorithm) {
+    protected Election createElectionAlgorithm() {
         Election le = null;
 
-        //TODO: use a factory rather than a switch
-        switch (electionAlgorithm) {
-        case 1:
-            le = new AuthFastLeaderElection(this);
-            break;
-        case 2:
-            le = new AuthFastLeaderElection(this, true);
-            break;
```

Comment:
similar to my comment above - if this were/is set wouldn't it be good to emit an error message?

Reply:
Agree. Throw an UnsupportedException here instead of removing the switch.

Reply:
Added back the switch statement and if cases are 1 and 2 then throw UnsupportedOperationException